src/huggingface_hub/file_download.py

```diff
@@ -961,6 +961,31 @@ def repo_folder_name(*, repo_id: str, repo_type: str) -> str:
     return REPO_ID_SEPARATOR.join(parts)
 
 
+def _check_disk_space(expected_size: Optional[int], target_dir: Union[str, Path]) -> None:
```

Comment:
```suggestion
def _check_disk_space(expected_size: int, target_dir: Union[str, Path]) -> None:
```

Your problem comes from the fact that you expect `int` or `None` as input value. Mypy tells you that if a None value is passed, the line `target_dir_free >= expected_size` will fail. This is how static typing is useful: mypy doesn't even need to run the code to tell you there is a bug. In this case, just change the input type and that's it! :)